### PR TITLE
enable AIRFLOW__CELERY__BROKER_URL_CMD when passwordSecretName is true

### LINKED ISSUE #39985: Can't use AIRFLOW__CELERY__BROKER_URL_CMD in the Helm Chart
### Official Helm Chart version

1.13.1 (latest released)

### Apache Airflow version

2.9.1

### Kubernetes Version

1.28.9

### Helm Chart configuration

```yaml
env:
  - name: AIRFLOW__CELERY__BROKER_URL_CMD
    value: |
      python -c '
      import urllib.parse, os
      pw = os.getenv("REDIS_PASSWORD")
      print(urllib.parse.quote_plus(f"redis://:${pw}@data-airflow-redis:6379/0", end="")
      '

envFrom:
  - name: REDIS_PASSWORD
    valueFrom:
      secretKeyRef:
        name: airflow-redis-password
        key: password

redis:
  enabled: true
  passwordSecretName: airflow-redis-password

enableBuiltInSecretEnvVars:
  AIRFLOW__CELERY__BROKER_URL: false
```

### Docker Image customizations

_No response_

### What happened

[Documentation suggests](https://airflow.apache.org/docs/helm-chart/stable/production-guide.html#built-in-secrets-and-environment-variables) that I can set my broker URL via a bash command. However, I get this error:
```
Error: execution error at (data-airflow/charts/airflow/templates/check-values.yaml:34:11): When using the internal redis of the chart and setting the value redis.passwordSecretName, you must also set the value data.brokerUrlSecretName.
```

### What you think should happen instead

Either the docs should be updated to reflect that developers can't use this env var, or the chart's value checks should be updated to allow setting only `redis.passwordSecretName` and not `redis.brokerUrlSecretName`.

### How to reproduce

The chart is rejecting my inputs as defined here: 
https://github.com/apache/airflow/blob/main/chart/templates/check-values.yaml#L33

### Anything else

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by jx2lee**
closes #39985

### CODE REVIEW

File: chart/templates/check-values.yaml [hunk 1]
```diff
@@ -30,8 +30,17 @@ The sole purpose of this yaml file is it to check the values file is consistent
   {{- if or (eq .Values.executor "CeleryExecutor") (eq .Values.executor "CeleryKubernetesExecutor") }}
     {{- if .Values.redis.enabled }}
 
-      {{- if and .Values.redis.passwordSecretName (not .Values.data.brokerUrlSecretName) }}
-        {{ required "When using the internal redis of the chart and setting the value redis.passwordSecretName, you must also set the value data.brokerUrlSecretName." nil }}
+      {{- if and .Values.redis.passwordSecretName }}
```
Comment (romsharon98) [inline chart/templates/check-values.yaml]:
```suggestion
      {{- if .Values.redis.passwordSecretName }}
```

File: chart/templates/check-values.yaml [hunk 2]
```diff
@@ -30,8 +30,17 @@ The sole purpose of this yaml file is it to check the values file is consistent
...
+        {{- $existedBrokerUrlCmd := false }}
+        {{- range .Values.env }}
+          {{- if eq .name "AIRFLOW__CELERY__BROKER_URL_CMD" }}
+            {{- $existedBrokerUrlCmd = true }}
```
Comment (romsharon98) [inline chart/templates/check-values.yaml:37]: you can add `{{- break -}}` after this line

File: helm_tests/other/test_redis.py [hunk 3]
```diff
@@ -143,7 +143,9 @@ def test_redis_by_chart_password(self, executor):
         self.assert_broker_url_env(k8s_obj_by_key)
 
     @pytest.mark.parametrize("executor", CELERY_EXECUTORS_PARAMS)
-    def test_redis_by_chart_password_secret_name_missing_broker_url_secret_name(self, executor):
+    def test_redis_by_chart_password_secret_name_missing_broker_url_secret_name_and_broker_url_cmd(
```
Comment (romsharon98) [inline helm_tests/other/test_redis.py:146]: add some test when you have `AIRFLOW__CELERY__BROKER_URL_CMD`

File: chart/templates/check-values.yaml (hunk 2 above)
Comment (jx2lee) [reply chart/templates/check-values.yaml:37]: thanks, done!

File: chart/templates/check-values.yaml (hunk 1 above)
Comment (jx2lee) [reply chart/templates/check-values.yaml]: thanks, done!

**Review by amoghrajesh — COMMENTED** [review]
Changes look ok but can we add a test specific to this change?

**Comment by eladkal** [comment]
@jx2lee can you add the unit test?

**Comment by jx2lee** [comment]
@amoghrajesh @eladkal 
yes, i've written unit test, but having a little problem in code. I'll push soon.

File: helm_tests/other/test_redis.py [hunk 4]
```diff
@@ -185,6 +185,32 @@ def test_redis_by_chart_password_secret_name(self, executor):
...
+                ],
+                "enableBuiltInSecretEnvVars": {
+                    "AIRFLOW__CELERY__BROKER_URL": False
+                }
+            },
+        )
+        k8s_obj_by_key = prepare_k8s_lookup_dict(k8s_objects)
+        created_redis_objects = SET_POSSIBLE_REDIS_OBJECT_KEYS & set(k8s_obj_by_key.keys())
+
+        assert created_redis_objects == SET_POSSIBLE_REDIS_OBJECT_KEYS - {
```
Comment (jx2lee) [inline helm_tests/other/test_redis.py:209]: What other values should i look at `assert`?

File: helm_tests/other/test_redis.py (hunk 3 above)
Comment (jx2lee) [reply helm_tests/other/test_redis.py:146]: @romsharon98 too late, sorry.
i pushed simple unit test!

File: helm_tests/other/test_redis.py (hunk 4 above)
Comment (jx2lee) [reply helm_tests/other/test_redis.py:209]: @amoghrajesh @romsharon98 could you check on this unit test codes, please?

**Review by romsharon98 — APPROVED** [review]
Sorry for the delay, the tests looks good.

**Review by potiuk — APPROVED** [review]

**Pull request merged**